StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1130UseLambdaSyntax.cs

```diff
@@ -50,12 +50,25 @@ public override void Initialize(AnalysisContext context)
         /// <summary>
         /// Gets the delegate parameter list from a method symbol and the argument index.
         /// </summary>
-        /// <param name="methodSymbol">The symbol containing information about the method invocation.</param>
+        /// <param name="symbol">The symbol containing information about the method invocation.</param>
         /// <param name="argumentIndex">The index of the argument containing the delegate.</param>
         /// <returns>A parameter list for the delegate parameters.</returns>
-        internal static ParameterListSyntax GetDelegateParameterList(IMethodSymbol methodSymbol, int argumentIndex)
+        internal static ParameterListSyntax GetDelegateParameterList(ISymbol symbol, int argumentIndex)
         {
-            var delegateType = (INamedTypeSymbol)methodSymbol.Parameters[argumentIndex].Type;
+            INamedTypeSymbol delegateType;
+            if (symbol is IMethodSymbol methodSymbol)
+            {
+                delegateType = (INamedTypeSymbol)methodSymbol.Parameters[argumentIndex].Type;
```

Comment:
❔ Is it possible for this cast to fail? For example, consider the following invalid code:

```csharp
void Method(int* data) => throw null;
void Caller() => Method(delegate { });
```

Reply:
I added a test, but this is not a concern because overload resolution just fails in those cases.